biomodels/src/main/java/org/reactome/release/ModelsTSVParser.java

```diff
@@ -0,0 +1,65 @@
+package org.reactome.release;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ModelsTSVParser {
+    private static final Logger logger = LogManager.getLogger();
+
+    private ModelsTSVParser() { }
+
+    /**
+     * Parses the contents of the models2pathways.tsv file, returning a Map of ReactomePathwayIds=[BioModelsIdentifiers,...]
+     * @param tsvFile -- String, path/to/models2pathways.tsv
+     * @return Map of ReactomePathwayIds and a List of BioModelsIdentifiers
+     */
+    public static Map<String, List<String>> parse(String tsvFile) {
+        Map<String, List<String>> pathwayToBiomodelsIds = new HashMap<>();
```

Comment:
To ensure there are no duplicates, a Set of Strings for the map's value might be better:
`Map<String, Set<String>> pathwayToBiomodelsIds = new HashMap<>()`

Reply:
This will require the code to get changed a bit more than what is suggested, so before I go ahead and make these changes, I'd like to point out that I have a commit for preventing duplicate identifiers being created: https://github.com/reactome/data-release-pipeline/pull/96/commits/d40d0f61065111aee17caf2c2fb3cf6d11a5c522

This uses the HashMap structure that is common in orthoinference to prevent duplicate instances from being created. If your overall goal is to prevent duplicate biomodels _instances_, then that has been handled in a different way. Otherwise, I will go ahead and make the suggested change

Reply:
I think this code change is still important.  The GKInstance created from an identifier won't be duplicated with the commit https://github.com/reactome/data-release-pipeline/commit/d40d0f61065111aee17caf2c2fb3cf6d11a5c522 but the List of String identifiers iterated over in the method `createBiomodelsDatabaseIdentifiers` could have the same String 2 or more times.  This would result in the unique GKInstance representing the BioModels identifier being added multiple times to the list returned from that method.  This could cause problems for the list of BioModels identifier instances in a pathway.  Using a Set of Strings prevents that.  IntelliJ should guide you wherever any changes are needed to change the data structure used from a list to a set after the type is changed on line 25.

Reply:
OK I've made the change. ModelsTSVParser now returns a Map<String, Set<String>>. This Set is then transformed into a List when creating the DatabaseIdentifiers for the BioModels IDs, which keeps the code mostly the same.